### PR TITLE
lnrpc: use initiator enum for pending channel open initiator 

### PR DESCRIPTION
**Pull request by carlaKC**
#4111 uses a bool for channel initiator status on `PendingChannel` which will result in ambiguity between channels which the remote party opened and those that we do not have a historical channel record of. 

This PR deprecates the field and uses the `Initiator` enum which allows for an `UNKNOWN` case.

### CODE REVIEW

File: lnrpc/rpc.proto [hunk 1]
```diff
@@ -2093,11 +2093,13 @@ message PendingChannelsResponse {
         */
         int64 remote_chan_reserve_sat = 7;
 
-        // True if we initiated opening the channel.
-        bool initiator = 8;
+        reserved 8;
```
Comment (joostjager) [inline lnrpc/rpc.proto:2096]: If this hasn't been released yet, I think we just stick with record 8 for the enum.

File: rpcserver.go [hunk 2]
```diff
@@ -2748,7 +2758,7 @@ func (r *rpcServer) PendingChannels(ctx context.Context,
 		case channeldb.ErrChannelNotFound:
 
 		case nil:
-			channel.Initiator = historical.IsInitiator
+			channel.Initiator = rpcInitiator(historical.IsInitiator)
```
Comment (joostjager) [inline rpcserver.go:2761]: Maybe it is good to explictly initialize `channel.Initiator` to unknown?

File: lnrpc/rpc.proto [hunk 3]
```diff
@@ -1483,6 +1483,13 @@ message ListChannelsResponse {
     repeated Channel channels = 11;
 }
 
+enum Initiator {
```
Comment (bhandras) [inline lnrpc/rpc.proto:1486]: Imho less generic name may work better. Eg. `ChannelInitiator`

Comment (carlaKC) [reply lnrpc/rpc.proto:1486]: We also use this enum for close initiators, so I'm not sure that fits? `ChannelInitiator` sounds like just opening to me?

**Comment by Roasbeef** [comment]
Heads up, this should only land _after_ the 0.10 tag.

**Comment by Roasbeef** [comment]
Discussed this briefly offline and an argument was made that this should land so we can get the breaking change in sooner rather than later (also it's relatively contained).

**Pull request merged**